ee/app/livechat-enterprise/server/lib/Helper.js

```diff
@@ -151,6 +154,21 @@ export const updatePredictedVisitorAbandonment = () => {
 	}
 };
 
+export const updateQueueInactivityTimeout = () => {
+	const queueAction = settings.get('Livechat_max_queue_wait_time_action');
+	const queueTimeout = settings.get('Livechat_max_queue_wait_time');
+	if (!queueAction || queueAction === 'Nothing') {
+		logger.debug('QueueInactivityTimer: No action performed (disabled by setting)');
+		return LivechatInquiry.unsetEstimatedInactivityCloseTime();
+	}
+
+	logger.debug('QueueInactivityTimer: Updating estimated inactivity time for queued items');
+	LivechatInquiry.getQueuedInquiries().forEach((inq) => {
```

Comment:
As far as I know, the [collection.update](https://docs.mongodb.com/v5.0/reference/method/db.collection.update/#db.collection.update) supports aggregation pipeline so you could update all queued inquiries at once, have you thought about it? In that case, you wouldn't need to loop all inquired queues and would execute just one query, something like this:
```
// { "date" : ISODate("2020-04-05T07:14:17.802Z"), "x" : "y" }
db.collection.updateMany(
  { date : { $exists : true } },
  [{ $set: { date: { $add: ["$date", 24*60*60000] } } }]
)
// { "date" : ISODate("2020-04-06T07:14:17.802Z"), "x" : "y" }
```

Think about it.

Reply:
Thanks for reminding me of this! Modified to use update pipes